### PR TITLE
Add release notes for 17.1 release

### PR DESCRIPTION
**Pull request by GaneshSPatil**

### CODE REVIEW

File: source/partials/next_release/_release.md.erb [hunk 1]
```diff
@@ -1,15 +1,29 @@
 #### API Enhancements
 
-* <%= link_to_commit 'f47d054171b07e2ce9c2efa9e02dde6979fc9c8f', "#{link_to_api 'encryption'} to encrypt secrets.", 1829 %>
+- <%= link_to_commit '73595c1d18f7455c5b7aac39dce319c987f0c231', 'Check Package Connection API (#3048)', 3048 %>
```
Comment (ketan) [inline source/partials/next_release/_release.md.erb]: You can additionally use `link_to_api` to add a link to the API docs. We do this for a couple of reasons —

* to improve SEO
* to help users discover links to the various sites we have

File: source/partials/next_release/_release.md.erb [hunk 2]
```diff
@@ -35,21 +49,23 @@
...
+* As of release 16.7, Java API based Go Plugins have been deprecated. Support for running these plugins will be removed in 17.2 (scheduled to be released in February 2017). Plugin developers are encouraged to upgrade their plugins to use [JSON message based plugin API.](https://plugin-api.gocd.io/current/tasks) to build plugins.
 
-    Below is the list of plugins that will stop working in 17.1.
+    Below is the list of plugins that will stop working in 17.2.
 
-    **Note:** The GoCD team has built a [shim](https://github.com/aresok/go-maven-poller/compare/master...arvindsv:make-it-work-with-gocd-17-1) that allows migration of all plugins with minimal change. We are actively working with plugin authors to ensure that these plugins are migrated over in time for 17.1.
+    **Note:** The GoCD team has built a [shim](https://github.com/aresok/go-maven-poller/compare/master...arvindsv:make-it-work-with-gocd-17-1) that allows migration of all plugins with minimal change. We are actively working with plugin authors to ensure that these plugins are migrated over in time for 17.2.
```
Comment (ketan) [inline source/partials/next_release/_release.md.erb]: Some of these plugins have had PRs submitted by @arvindsv. You'll need to check each plugin if they have cut a new release with the said PR, and ensure that we mark these plugins saying something along the lines of "this plugin will now work with 17.2"

File: source/partials/next_release/_release.md.erb (hunk 1 above)
Comment (varshavaradarajan) [inline source/partials/next_release/_release.md.erb]: Check Package Connection is an internal API. Please remove.

File: source/partials/next_release/_release.md.erb [hunk 3]
```diff
@@ -1,15 +1,29 @@
 #### API Enhancements
 
-* <%= link_to_commit 'f47d054171b07e2ce9c2efa9e02dde6979fc9c8f', "#{link_to_api 'encryption'} to encrypt secrets.", 1829 %>
+- <%= link_to_commit '73595c1d18f7455c5b7aac39dce319c987f0c231', 'Check Package Connection API (#3048)', 3048 %>
+- <%= link_to_commit '86729312550d47c7268c615fc0926bd6d37d4d7e', 'Check Repository Connection API. (#2953)', 2954 %>
```
Comment (varshavaradarajan) [inline source/partials/next_release/_release.md.erb]: Check Repository Connection is an internal API. Please remove

File: source/partials/next_release/_release.md.erb [hunk 4]
```diff
@@ -1,15 +1,29 @@
...
+- <%= link_to_commit '041572cf78caecb32157d7a68fdca183dcff6674', 'Clone notification API endpoint.', 2954 %>
+- <%= link_to_commit '2789aa8f691a860c7908aabb42cacc99a7333031', 'Allow template admin to view and edit templates via API.', 2967 %>
+- <%= link_to_commit '3c001b1d376263e964495723fb76a4771ca8a705', 'API V2 of stage status notification.', 3030 %>
```
Comment (varshavaradarajan) [inline source/partials/next_release/_release.md.erb]: This is not an api change. The commit said it was version 2 of the notification endpoint. This is for the notification plugins.

File: source/partials/next_release/_release.md.erb [hunk 5]
```diff
@@ -1,15 +1,29 @@
...
+- <%= link_to_commit '467cf5f1c602144ab3e19fb95f5ab512c309d2de', 'Encryption API: Use a representer to render the response along with links. (#3031)', 3031 %>
```
Comment (varshavaradarajan) [inline source/partials/next_release/_release.md.erb]: Remove ```Use a representer to render the response along with links. (#3031)```

File: source/partials/next_release/_release.md.erb [hunk 6]
```diff
@@ -1,15 +1,29 @@
...
+- <%= link_to_commit 'ea7489d6aa797e4a63bb122cfe821e09ac2c3b9d', 'Disallow template admins to delete templates via API in accordance with the behaviour in the UI.', 3041 %>
```
Comment (varshavaradarajan) [inline source/partials/next_release/_release.md.erb]: No need for this. The message ```Allow template admin to view and edit templates via API.``` above takes care of this

File: source/partials/next_release/_release.md.erb [hunk 7]
```diff
@@ -1,15 +1,29 @@
...
 
 #### Other Improvements
 
-* <%= link_to_commit 'xxx', 'YYY.', 123 %>
+* <%= link_to_commit '2ae7fdb72e31b2868cee8a3d757402e2e0e40456', 'Full config save optimization', 2912 %>
```
Comment (varshavaradarajan) [inline source/partials/next_release/_release.md.erb]: Add this under the heading Performance improvements.

File: source/partials/next_release/_release.md.erb [hunk 8]
```diff
@@ -1,15 +1,29 @@
...
+* <%= link_to_commit '0802c7ede115efd5c71bfcf5f5056261918f43e7', 'Adding CPU time in api/support for the threads.', 3006 %>
+* <%= link_to_commit 'c749c47a12df283c2e333b9cc918e161aa6c1803', 'Add a link to pipeline settings in the pipeline history page. (#2939)', 2939 %>
 
 #### Bug Fixes
 
 * <%= link_to_commit 'e9611f83798bd5eddd62326f0827e5055519bd4a', 'OS Detection failing for elastic agents.', 3060 %>
 * Multiple UI fixes and tweaks on the new theme that was introduced in 16.12.
+* <%= link_to_commit 'e9611f83798bd5eddd62326f0827e5055519bd4a', 'Refresh ElasticAgent-OperatingSystem and ElasticAgent-UsableSpace while initializing the runtimeInfo (#3060)', 3078 %>
+* <%= link_to_commit '4b40218b5f8a63b3cf528d1f7c0613ee1c886dad', 'Rewrite all URLs to gocd.io (#3057)', 3057 %>
+* <%= link_to_commit '31ab33de0ef3f0b3718cfc095e12a4665ac4319b', 'Change README to use gocd.io' %>
```
Comment (varshavaradarajan) [inline source/partials/next_release/_release.md.erb]: I think you can remove this. This isn't a bug.

File: source/partials/next_release/_release.md.erb [hunk 9]
```diff
@@ -35,21 +49,23 @@
...
+* As of release 16.12, git versions older than 1.9 are deprecated. Support for git versions older than 1.9 will be stopped in 17.2 (scheduled to be released in February 2017). Users are recommended to upgrade git on the GoCD server and agents (if git is used as a material). We are aware that the linux distributions have an older version of git installed by default. Please refer to [this](https://www.softwarecollections.org/en/scls/rhscl/git19/) and/or [this](http://askubuntu.com/questions/571549/git-1-7-9-5-upgrade-to-current-release-of-git-2-x-on-ubuntu-12-04/571633#571633) to upgrade git.
 * As of release 16.12, version 1 of the <%= link_to_api 'get-all-plugin-info', 'Plugin Info API' %> is deprecated. This version of the API will be removed in 17.2 release (scheduled to be released in Feb 2017). Version 2 of the API is available, and users are encouraged to use it.
+* As of release 17.1, version 2 of the <%= link_to_api 'template-config', 'Templates API' %> is deprecated. This version of the API will be removed in 17.3 release (scheduled to be released in March 2017). Version 3 of the API is available, and users are encouraged to use it.
```
Comment (varshavaradarajan) [inline source/partials/next_release/_release.md.erb]: we give a 3 month deprecation period for APIs. Version 2 will be removed in 17.4 (Scheduled for April 2017)

File: source/partials/next_release/_release.md.erb (hunk 7 above)
Comment (varshavaradarajan) [inline source/partials/next_release/_release.md.erb]: Have the section Performance improvements to be at the beginning.

**Pull request merged**